### PR TITLE
fix: Fix bottom padding in modal

### LINKED ISSUE #5576: suite-native: BottomModal padding SafeArea bug
There is text going outside of screen for BottomModal:
Screenshot 2022-06-07 at 16 31 53

You should solve this probably using by using `useSafeAreaInsets` https://github.com/th3rdwave/react-native-safe-area-context#usesafeareainsets and adding bottom value to our bottom padding.

### PR DESCRIPTION
**Pull request by juriczech**
Screenshots:
- IOS https://tppr.me/xq0MP
- Android https://tppr.me/AlElJ

Closes: [#5576](https://github.com/trezor/trezor-suite/issues/5576)

### CODE REVIEW

File: packages/suite-native-atoms/src/Modal/BottomModalContainer.tsx [hunk 1]
```diff
@@ -18,8 +18,14 @@ const modalWithOverlayStyle = prepareNativeStyle(utils => ({
...
 export const BottomModalContainer = ({ children, isVisible, onClose }: ModalProps) => {
     const { applyStyle } = useNativeStyles();
+
     return (
-        <RNModal transparent visible={isVisible} onRequestClose={onClose}>
+        <RNModal
+            transparent
+            visible={isVisible}
+            onRequestClose={onClose}
+            style={{ paddingBottom: 100 }}
```
Comment (Nodonisko) [inline packages/suite-native-atoms/src/Modal/BottomModalContainer.tsx]: are you sure this should be here?

File: packages/suite-native-atoms/src/Modal/BottomModal.tsx [hunk 2]
```diff
@@ -17,10 +18,14 @@ type BottomModalProps = {
     onBackArrowClick?: () => void;
 };
 
-const modalWrapperStyle = prepareNativeStyle(utils => ({
+type WrapperStyleProps = {
+    insets: EdgeInsets;
```
Comment (Nodonisko) [inline packages/suite-native-atoms/src/Modal/BottomModal.tsx]: I would prefer to pass just `insetBottom: number` to make this little bit simplier

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: packages/suite-native-atoms/src/Modal/BottomModalContainer.tsx (hunk 1 above)
Comment (juriczech) [reply packages/suite-native-atoms/src/Modal/BottomModalContainer.tsx]: Fuuuck! Good catch. I was just testing different stuff and forgot to remove it.

File: packages/suite-native-atoms/src/Modal/BottomModal.tsx (hunk 2 above)
Comment (juriczech) [reply packages/suite-native-atoms/src/Modal/BottomModal.tsx]: [5ca9f07](https://github.com/trezor/trezor-suite/pull/5579/commits/5ca9f079c5a74efd47d083b6b62fc98ebe7fe64e)

File: packages/suite-native-atoms/src/Modal/BottomModalContainer.tsx (hunk 1 above)
Comment (juriczech) [reply packages/suite-native-atoms/src/Modal/BottomModalContainer.tsx]: [5ca9f07](https://github.com/trezor/trezor-suite/pull/5579/commits/5ca9f079c5a74efd47d083b6b62fc98ebe7fe64e)

**Review by bouchja1 — APPROVED** [review]

File: packages/suite-native-atoms/src/Modal/BottomModal.tsx [hunk 3]
```diff
@@ -49,13 +54,18 @@ export const BottomModal = ({
...
+    const insets = useSafeAreaInsets();
 
     const handleCloseModal = () => onVisibilityChange(false);
 
     return (
         <BottomModalContainer isVisible={isVisible} onClose={handleCloseModal}>
-            <Box style={applyStyle(modalWrapperStyle)}>
+            <Box
+                style={applyStyle(modalWrapperStyle, {
+                    insetBottom: Math.max(insets.bottom, utils.spacings.md),
```
Comment (Nodonisko) [inline packages/suite-native-atoms/src/Modal/BottomModal.tsx]: Please move Max to style, that's is place where this style related logic belongs and also it don't needs to be recalculated during every rerender

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: packages/suite-native-atoms/src/Modal/BottomModal.tsx (hunk 3 above)
Comment (juriczech) [reply packages/suite-native-atoms/src/Modal/BottomModal.tsx]: [ac0c421](https://github.com/trezor/trezor-suite/pull/5579/commits/ac0c421238ff07a6aeb2fa6e7b2c120bdb5557ac)

File: packages/suite-native/package.json [hunk 4]
```diff
@@ -27,8 +27,8 @@
         "react-native": "0.68.2",
         "react-native-gesture-handler": "^2.4.2",
         "react-native-reanimated": "^2.8.0",
-        "react-native-screens": "^3.13.1",
-        "react-native-safe-area-context": "^4.2.5"
+        "react-native-safe-area-context": "^4.3.1",
```
Comment (Nodonisko) [inline packages/suite-native/package.json:30]: Any particular reason to update these deps?

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**